doc/adr/0009-time-graph-search-functionality.md

```diff
@@ -0,0 +1,133 @@
+
+Currently, the trace-server-protocol documentation for [getStates](https://eclipse-cdt-cloud.github.io/trace-server-protocol/#tag/TimeGraph/operation/getStates) does not involve any query parameters for searching or filtering states. The '/states' endpoint takes in the parameters 'requested_timerange' and 'requested_items'. The following is an example:
+```
+{
+  "parameters": {
+    "requested_timerange": {
+      "start": 111111111,
+      "end": 222222222,
+      "nbTimes": 1920
+    },
+    "requested_items": [
+      1,
+      2
+    ]
+  }
+}
+```
+- The array 'requested_times' is the explicit array of requested sample times. The array 'requested_items' is the list of entryId's being requested.
+
+But it is possible to augment the request body by adding 'regex_map_filters' and 'full_search'. These parameters are currently accepted by the time graph data provider - and they can be used to support search/filter expressions. Here's an example where we can use these parameters to search for states with the label "openat":
+```
+{
+  "parameters": {
+    "requested_timerange": {
+      "start": 111111111,
+      "end": 222222222,
+      "nbTimes": 1920
+    },
+    "requested_items": [
+      1,
+      2
+    ],
+    "regex_map_filters": {1: ["openat"]},
+    "full_search": true
+  }
+}
+```
+-   “regex_map_filters” is an array with requests for searching. The  key  of  this  map  can  be \"1\" (means  DIMMED) or \"4\" (means  EXCLUDED) and  the  value  is  an  array  of  the  desired  search  query (e.g. {\"1\": [\"openat\", \"duration>10ms\"]})"
+-  The “full_search” key is an optional parameter that enables the full_seach (deep search) or not. 
+   - If true then the data provider is expected to perform a full search of elements that occur between the usual elements found at the requested times (e.g. in gaps).  If any element matches the specified regex filter, at least one such  matching element per gap should be returned. It is not expected to return  all matching elements found in a single gap, for performance reasons.  
+   - In Trace Compass when we perform a graph search, we first do a normal non-full-search query for all the visible rows, in order to quickly show something to the user. Then we perform a full-search in the background. Because this process can be very slow, we do it one row at a time and update the graph incrementally as each row's fine-grained result comes back.
+
+
+I've attached a link to a prototype in trace-extension that makes use of the aforementioned parameters to  search time events in time graph views:
+- [https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886](https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886)
+- This prototype adds a search-box to the timegraph-output-component to search for events. The filter is applied and unsuccessful time events are dimmed.
```

Comment:
This might not be the final UI, but it gives the idea how to achieve it.

Reply:
Done